File: Problem 028/Solution 28.rb
# Solves http://projecteuler.net/problem=28

sum = 0
corner = [ "0", "0", "0", "0" ]

for x in (3..1001).step(2)
	corner[3] = x*x
	corner[2] = corner[3] - (x-1)
	corner[1] = corner[2] - (x-1)
	corner[0] = corner[1] - (x-1)

	for a in (0..3)
		sum += corner.fetch(a)
	end
end

print "Answer: ", sum, "\n"

File: Problem 028/Solution 28.c
/* Solves http://projecteuler.net/problem=28 */

#include <stdio.h>

int main()
{
	int i;
	int Corner[4];
	int sum = 1;
	int a;

	for(i = 3; i <= 1001; i += 2)
	{
		/* This calculates each corner as we go down the grid */
		Corner[3] = i*i;
		Corner[2] = Corner[3] - (i-1);
		Corner[1] = Corner[2] - (i-1);
		Corner[0] = Corner[1] - (i-1);

		/* Calculate the sum */
		for(a = 0; a < 4; ++a)
			sum += Corner[a];
	}

	printf("Answer: %d\r\n", sum);

	return 0;
}
